ui/src/views/sql-query-editor/index.tsx

```diff
@@ -36,6 +37,33 @@ const QueryEditor: React.FC = () => {
     }
   })
 
+  const projectionChanged = (columns: ColumnInfo[]) => {
+    const newProjection = columns.map(c => c.name)
+    const previousProjection = projection.map(c => c.name)
+
+    let changed = false
+    newProjection.forEach(p => {
+      if (!previousProjection.includes(p)) {
+        changed = true
+      }
```

Comment:
you could potentially exit this loop early if you've found a change. So if there's a change in the first item, you don't need to iterate over all the remaining items

Reply:
it was improved here [5ad57eb](https://github.com/mergestat/mergestat/pull/785/commits/5ad57eb99c8274658feb61371f637044cfa689f3) 👍